src/index.js

```diff
@@ -1,42 +1,102 @@
+  return subtotal
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
-  // ITERATION 3
-  //... your code goes here
+  const products = document.querySelectorAll('.product');
+  let total = 0
+  products.forEach(product =>{
+    total += updateSubtotal(product)
+  })
+  const totalValueElement = document.getElementById("total-value")
+  totalValueElement.innerText = total
+  return total
 }
 
-// ITERATION 4
-
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  const target = event.currentTarget.parentNode.parentNode;
+  target.remove()
 }
 
-// ITERATION 5
+function createProduct(event) {
+  let target = event.currentTarget.parentNode.parentNode
+  if(!target.querySelector("input").value){
+    target.querySelector("input").style.border = "2px solid red"
+    return;
+  }
+  target.querySelector("input").style.border = "none"
+  let newProduct = document.createElement("tr")
+  newProduct.classList.add("product")
+
+  let nameTd = document.createElement("td")
+  nameTd.classList.add("name")
+  let nameSpan = document.createElement("span")
+  nameSpan.innerText = target.querySelector("input").value
+  nameTd.appendChild(nameSpan)
+
+  let priceTd = document.createElement("td")
+  priceTd.classList.add("price")
+  let priceSpan = document.createElement("span")
+  priceSpan.innerText = target.querySelector('input[type="number"]').value + ".00"
+  priceTd.innerText = "$"
+  priceTd.appendChild(priceSpan)
+  
+  let quantityTd = document.createElement("td")
+  quantityTd.classList.add("quantity")
+  let quantityInput = document.createElement("input")
+  quantityInput.type = "number"
+  quantityInput.min = 0
+  quantityInput.value = 0
+  quantityInput.placeholder = "Quantity"
+  quantityTd.appendChild(quantityInput)
+
+  let subtotalTd = document.createElement("td")
+  subtotalTd.classList.add("subtotal")
+  let subtotalSpan = document.createElement("span")
+  subtotalTd.innerText = "$"
+  subtotalSpan.innerText = "0"
+  subtotalTd.appendChild(subtotalSpan)
+
+  let removeButtonTd = document.createElement("td")
+  removeButtonTd.classList.add("action")
+  let removeButton = document.createElement("button")
+  removeButton.classList.add("btn")
+  removeButton.classList.add("btn-remove")
+  removeButton.innerText = "Remove"
+  removeButtonTd.appendChild(removeButton)
+  removeButton.addEventListener("click", function(event){
+    removeProduct(event)
+  })
+
+  newProduct.appendChild(nameTd)
+  newProduct.appendChild(priceTd)
+  newProduct.appendChild(quantityTd)
+  newProduct.appendChild(subtotalTd)
+  newProduct.appendChild(removeButtonTd)
+
```

Comment:
A little tip: you could save a looooot of lines using the method cloneNode 

https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode